jira/client.py

```diff
@@ -771,6 +784,9 @@ def _fetch_pages(
                     and (total is None or page_start < total)
                     and len(next_items_page) == page_size
                 ):
+                    page_params = (
+                        params.copy() if params else {}
+                    )  # Hack necessary for mock-calls to not change
```

Comment:
If omitted the mock calls will point to the same page-params object, which causes retroactive changes to already fired mock calls.

Reply:
hmm, the thing is this is already done at the top of the function where we actually use the batch size...  maybe it needs to be `page_params = page_params.copy()`

I'll need to step by step run your test through a debugger to make sure this is what we want

Reply:
hmm, not a huge fan of this, but it works and doesn't really do much harm so let's keep it in